angular-workspace/projects/ni/nimble-angular/table-column/anchor/nimble-table-column-anchor-navigation-guard.directive.ts

```diff
@@ -0,0 +1,39 @@
+import { Directive, HostListener, Input } from '@angular/core';
+import type { TableColumnAnchorCellView } from '@ni/nimble-components/dist/esm/table-column/anchor/cell-view';
+
+export type ClickDelegate = (rowRecordId: string | undefined) => boolean;
+
+/**
+ * Directive to allow client to intercept anchor clicks and do router navigation instead of
+ * letting the default action handle the navigation.
```

Comment:
Can you make it dead clear what the app should do, i.e. the behavior of returning true vs false and a short description of how this delegate is expected to be used